klarna-on-demand/src/test/java/com/klarna/ondemand/UrlHelperTest.java

```diff
@@ -0,0 +1,84 @@
+package com.klarna.ondemand;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.rule.PowerMockRule;
+
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.util.Locale;
+
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(emulateSdk = 18)
+@PrepareForTest({Context.class, Locale.class})
+@PowerMockIgnore({ "org.mockito.*", "org.robolectric.*", "android.*" })
+public class UrlHelperTest {
+
+    private String token = "my_token";
+
+    @Rule
+    public final PowerMockRule rule = new PowerMockRule();
+
+    @Before
+    public void beforeEach() {
+        mockStatic(Context.class);
+        when(Context.getApiKey()).thenReturn("test_skadoo");
+    }
+
+    //region .registrationUrl
+    @Test
+    public void registrationUrl_ShouldReturnPlaygroundUrl_WhenTokenIsForPlayground() {
+        Assert.assertTrue(UrlHelper.registrationUrl().startsWith("https://inapp.playground.klarna.com/registration/new"));
+    }
+
+    @Test
+    public void registrationUrl_ShouldReturnProductionUrl_WhenTokenIsForProduction() {
+        mockStatic(Context.class);
+        when(Context.getApiKey()).thenReturn("skadoo");
+
+        Assert.assertTrue(UrlHelper.registrationUrl().startsWith("https://inapp.klarna.com/registration/new"));
+    }
+
+    @Test
+    public void registrationUrl_ShouldReturnUrlWithSwedishLocale_WhenLocaleIsSwedish() {
```

Comment:
The test names don't make sense anymore, now that we are no longer talking about Swedish.
Also, the inherent problem remains. When the locale is set to "Swedish" we test to see that the generated URLs reflect it, but we have no test that verifies that if the locale **is not** "Swedish", the URL **does not** contain it.